index.d.ts

```diff
@@ -157,9 +159,14 @@ export interface ModifierInvocation extends BaseASTNode {
 }
 export interface FunctionDefinition extends BaseASTNode {
   type: 'FunctionDefinition';
-  name: string;
+  name?: string;
   parameters: ParameterList;
-  body: Block | null;
+  modifiers: ModifierInvocation[];
+  stateMutability?: 'pure' | 'constant' | 'payable' | 'view'
+  visibility: 'default' | 'external' | 'internal' | 'public' | 'private';
+  isConstructor: boolean;
+  returnParameters?: ParameterList;
+  body?: Block;
```

Comment:
@Leeleo3x Can you give an example when `body` is undefined?

Reply:
```
contract ERC20Interface {

    function totalSupply() public view returns (uint);

    function balanceOf(address tokenOwner) public view returns (uint balance);

    function allowance(address tokenOwner, address spender) public view returns (uint remaining);

    function transfer(address to, uint tokens) public returns (bool success);

    function approve(address spender, uint tokens) public returns (bool success);

    function transferFrom(address from, address to, uint tokens) public returns (bool success);


    event Transfer(address indexed from, address indexed to, uint tokens);

    event Approval(address indexed tokenOwner, address indexed spender, uint tokens);

}
```